### ISSUE TITLE
False-positive "Cannot borrow immutable local variable as mutable"

### ISSUE THREAD

**Issue by Seeker14491** [intellij-rust/intellij-rust#3914]
The following code compiles, but shows an error in the plugin (shown in the comment)

```rust
fn main() {
    f(|x| &mut x.0);
            // ^^^ Cannot borrow immutable local variable `x.0` as mutable
}

fn f(_g: impl Fn(&mut (u8,)) -> &mut u8) {}
```

**Plugin version:** 0.2.98.2124-191
